File: web-development-101/javascript-exercises-test-driven/javascript-exercises/calculator/calculator.js
function add (a, b) {
	return a + b;
}

function subtract (a, b) {
	return a - b;
}

function sum (a) {
	let summation = a.reduce((accumulator, e) => accumulator + e, 0);
	return summation;
}

function multiply (a) {
	let product = a.reduce((accumulator, e) => accumulator * e, 1);
	return product;
}

function power(base, exponent) {
	return base ** exponent;
}

function factorial(n) {
	fact = 1;
	for(let i = 2; i <= n; ++i)
		fact *= i;
	return fact;
}

module.exports = {
	add,
	subtract,
	sum,
	multiply,
    power,
	factorial
}

File: web-development-101/learn-ruby-test-driven/02_calculator/calculator.rb
#write your code here
def add(a, b)
    a + b
end

def sum(elements)
    sum = 0
    elements.each{|e| sum = sum + e}
    sum
end

def subtract(a, b)
    a - b
end

def multiply(a, *elements)
    product = 1.0
    for e in elements do
        product = product * e
    end
    product * a
end

def factorial(n)
    f = 1;
    while n > 0
        f = f * n;
        n = n - 1;
    end
    f
end

def power(b, n)
    b ** n
end
